### DISCUSSION TITLE
训练自己的识别模型无法转nb模型

### DISCUSSION THREAD

**Discussion by Zomcxj** [PaddlePaddle/PaddleOCR#12340] [Q&A]
PaddleOCR 2.7.1
Paddlelite13rc0  
自训练det模型可转
训练自己的数据集使用key_ppocr_v1.txt或自定义字典txt   
v2 v3 v4 训练加载官方预训练模型    
测试的配置文件ch_PP-OCRv2_rec.yml ch_PP-OCRv3_rec.yml   ch_PP-OCRv4_rec.yml ch_PP-OCRv3_rec_distill.yml
infer模型转nb时可以加载1. Model is successfully loaded!  但无法优化并保存nb
ch_PP-OCRv4_rec_distill.yml直接无法训练，加载文字标签报错

**Comment by Zomcxj** [comment]
只能降版本Paddlelite2.12可以

**Comment by Alanhzl** [comment]
> 只能降版本Paddlelite2.12可以

你好，我想请问一下，我现在用的也是Paddlelite2.12，自己编译的opt工具，v2识别训练模型能转infer赚nb，但是v3能转infer不能转nb，报错
`Check failed: it != attrs().end(): No attributes called beta found for swish`
请问你用Paddlelite2.12，将v3的识别训练模型转到nb是可以的吗？

**Comment by Zomcxj** [comment]
可以的

**Comment by Alanhzl** [comment]
> 可以的

请问能说说是怎么操作的吗？我对比一下看看差别在哪
`python3 tools/export_model.py -c configs/rec/PP-OCRv3/ch_PP-OCRv3_rec_distillation.yml -o Global.pretrained_model="./pretrain_models/ch_PP-OCRv3_rec_train/best_accuracy" Global.save_inference_dir="./pretrain_models/ch_PP-OCRv3_rec_train/inference"
`
`./opt --model_file=./rec/rec_v3/Student/inference.pdmodel  --param_file=./rec/rec_v3/Student/inference.pdiparams  --optimize_out=./rec/rec_v3/Student/ch_ppocr_v3.0_rec_opt --valid_targets=arm  --optimize_out_type=naive_buffer
`
上面是我转模型的命令，前天意外中用了V2的配置文件去导出V3的推理模型，最后成功转成了nb模型。用configs/rec/PP-OCRv3/ch_PP-OCRv3_rec_distillation.yml这个就不行。
